src/main/groovy/org/transmartproject/batch/batchartifacts/CollectDatasetStatisticsListener.groovy

```diff
@@ -0,0 +1,34 @@
+package org.transmartproject.batch.batchartifacts
+
+import groovy.util.logging.Slf4j
+import org.springframework.batch.core.annotation.BeforeProcess
+import org.springframework.batch.item.ExecutionContext
+import org.springframework.beans.factory.annotation.Value
+import org.transmartproject.batch.highdim.datastd.DataPoint
+import org.transmartproject.batch.highdim.datastd.OnlineStatisticsCalculator
+
+/**
+ * Collect data statistics. E.g. minimal value.
+ */
+@Slf4j
+class CollectDataSetStatisticsListener {
+
+    public static final String DATA_SET_STATISTICS = 'dataSetStatistics'
+
+    @Value('#{jobExecution.executionContext}')
+    private ExecutionContext jobExecutionContext
+
+    @BeforeProcess
+    void beforeProcess(DataPoint dataPoint) {
+        OnlineStatisticsCalculator statistics
+        if (jobExecutionContext.containsKey(DATA_SET_STATISTICS)) {
+            statistics = jobExecutionContext.get(DATA_SET_STATISTICS)
+        } else {
+            log.debug('Create new data statistic bean and add it to the job context.')
+            statistics = new OnlineStatisticsCalculator()
+            jobExecutionContext.put(DATA_SET_STATISTICS, statistics)
+        }
+
+        statistics.push(dataPoint.value)
+    }
```

Comment:
Several problems here:

* Spends CPU cycles calculatign the variance and mean on the 1st step, when you only need to calculate the global mean value.
* For every datapoint, it fetches data a value from the context; which is also wasteful.
* Because the context is written to only on the first call to `beforeProcess`, Spring Batch will not know that the job execution context actually changed in subsequent calls (the `dirty` flag is not set) and may not serialize it. The best solution IMO would be to implement `ItemStream`. This would still require the promotion of the minimum value to the job context at the end of the step, but it would minimize writes to global state.